src/main/scala/ch/epfl/bluebrain/nexus/kg/async/ProjectViewCoordinatorActor.scala

```diff
@@ -156,92 +162,104 @@ private abstract class ProjectViewCoordinatorActor(viewCache: ViewCache[Task])(
-      children += view -> ref
+      logStart(view, s"restartOffset for projections progress: '${restartOffsetViews.map(_.progressId)}'")
+      children += view -> startCoordinator(view, project, restartOffsetViews)
     }
 
     {
       case ViewsAddedOrModified(_, restartOffset, views) =>
-        views.foreach {
+        val _ = views.map {
           case view if !children.keySet.exists(_.id == view.id) => startView(view, restartOffset)
-          case view: IndexedView =>
+          case view =>
             children
-              .collectFirst {
-                case (v: IndexedView, ref) if v.id == view.id && v.ref == view.ref && v.rev != view.rev =>
-                  v -> ref
-              }
+              .collectFirst { case (v, coordinator) if v.id == view.id && v.rev != view.rev => v -> coordinator }
               .foreach {
-                case (old, ref) =>
+                case (old, coordinator) =>
                   startView(view, restartOffset)
                   logStop(old, s"a new rev of the same view is going to be started, restartOffset '$restartOffset'")
-                  stopView(old, ref)
+                  stopView(old, coordinator)
               }
-          case _ =>
         }
 
       case ViewsRemoved(_, views) =>
         children.filterKeys(v => views.exists(_.id == v.id)).foreach {
-          case (v, ref) =>
+          case (v, coordinator) =>
             logStop(v, "removed from the cache")
-            stopView(v, ref)
+            stopView(v, coordinator)
         }
 
       case ProjectChanges(_, newProject) =>
         context.become(initialized(newProject))
         children.foreach {
-          case (view, ref) =>
+          case (view, coord) =>
             logStop(view, "project changed")
-            stopView(view, ref).map(_ => self ! ViewsAddedOrModified(project.uuid, restartOffset = true, Set(view)))
+            stopView(view, coord).map(_ => self ! ViewsAddedOrModified(project.uuid, restartOffset = true, Set(view)))
         }
 
-      case RestartView(_, view) =>
-        children.find { case (v, _) => v.id == view.id }.foreach {
-          case (v, ref) =>
-            logStop(v, "restart triggered from client")
-            stopView(v, ref, deleteIndices = false)
+      case RestartView(uuid, viewId) =>
+        val _ = children.findBy[IndexedView](viewId) match {
+          case Some((view, coordinator)) =>
+            sender() ! Option(Ack(uuid))
+            logStop(view, "restart triggered from client")
+            stopView(view, coordinator, deleteIndices = false)
               .map(_ => self ! ViewsAddedOrModified(project.uuid, restartOffset = true, Set(view)))
+          case None => sender() ! None
+        }
+
+      case RestartProjections(uuid, viewId) =>
+        val _ = children.findBy[CompositeView](viewId) match {
+          case Some((view, coordinator)) =>
+            sender() ! Option(Ack(uuid))
+            logStop(view, "restart triggered from client")
```

Comment:
We should probably log differently when a restart was caused by scheduled and when it was caused by client. It will probably make the debugging a bit easier.

Reply:
Yeah, I'll add something to this logging in the next PR